packages/notched-outline/index.tsx

```diff
@@ -91,49 +89,45 @@ export default class NotchedOutline extends React.Component<
     return classnames('mdc-notched-outline', Array.from(classList), className);
   }
 
-  get adapter() {
+  get adapter(): MDCNotchedOutlineAdapter {
     return {
-      getWidth: () => this.outlineElement_.current ? this.outlineElement_.current.offsetWidth : 0,
-      getHeight: () => this.outlineElement_.current ? this.outlineElement_.current.offsetHeight : 0,
       addClass: (className: string) =>
         this.setState({classList: this.state.classList.add(className)}),
       removeClass: (className: string) => {
         const {classList} = this.state;
         classList.delete(className);
         this.setState({classList});
       },
-      setOutlinePathAttr: (value: string) => {
-        if (this.pathElement_.current) {
-          this.pathElement_.current.setAttribute('d', value);
-        }
+      setNotchWidthProperty: (width: number) => {
+        this.setState({
+          actualWidth: width,
```

Comment:
I'm confused why there is `actualWidth` and `notchWidth`. `notchWidth` is supposed to come from text-field, and then `actualWidth` is set from the foundation, which adds any padding. Is this correct?